docs/source/data/subscriptions.mdx

```diff
@@ -0,0 +1,508 @@
+---
+title: Subscriptions
+description: Persistent GraphQL operations
+---
+
+import {ExpansionPanel} from 'gatsby-theme-apollo-docs';
+
+> **Subscriptions are not supported in [Apollo Federation](https://www.apollographql.com/docs/federation/).**
+
+**Subscriptions** are GraphQL operations that maintain an active connection between client and server. This enables your server to _push_ updates to a subscription's result as they occur, instead of requiring your client to _poll_ for updates.
+
+Because subscriptions use persistent two-way communication, they usually use [the WebSocket protocol](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) instead of HTTP. To support this, Apollo Server lets you [set a subscription-specific endpoint](#setting-a-subscription-endpoint) that's separate from the default endpoint for queries and mutations.
+
+You can use subscriptions with the core `apollo-server` library, or with any of Apollo Server's supported [middleware integrations](#using-with-middleware-integrations).
+
+> **Important:** Compared to queries and mutations, subscriptions are significantly more complex to implement. Before you begin, [confirm that your use case requires subscriptions](https://www.apollographql.com/docs/react/data/subscriptions/#when-to-use-subscriptions).
+
+## Schema definition
+
+You define your schema's supported subscriptions as fields of the `Subscription` type:
+
+```graphql
+type Subscription {
+  postCreated: Post
+}
+```
+
+The `postCreated` subscription will update its value whenever a new `Post` is created on the backend, thus pushing the `Post` to subscribing clients.
+
+Clients can execute this subscription with a GraphQL string like this:
+
+```graphql
+subscription PostFeed {
+  postCreated {
+    author
+    comment
+  }
+}
+```
+
+## Setting a subscription endpoint
+
+Because subscriptions use WebSocket instead of HTTP, Apollo Server uses a _second_ GraphQL endpoint specifically for subscriptions. This endpoint uses the `ws` protocol instead of `http`.
+
+By default, the subscription endpoint's path matches the path of your primary GraphQL endpoint (`/graphql` if not set). You can specify a different path for your subscription endpoint like so:
+
+```js
+const server = new ApolloServer({
+  subscriptions: {
+    path: '/subscriptions'
+  },
+  // ...other options...
+}));
+```
+
+## Resolving a subscription
+
+Resolvers for subscriptions differ from those for queries and mutations. Specifically, subscription resolvers are _objects_ that define a `subscribe` function:
+
+```js:title=index.js
+const resolvers = {
+  Subscription: {
+    postCreated: {
+      // More on pubsub below
+      subscribe: () => pubsub.asyncIterator(['POST_CREATED']),
+    },
+  },
+  // ...other resolvers...
+};
+```
+
+The `subscribe` function must return an object of type `AsyncIterable`, part of Apollo Server's pub/sub API (described below).
```

Comment:
`AsyncIterable` isn't really part of Apollo Server's API, but defined in ES2018. Maybe we could mention that it's a standard JavaScript API that is used to return a stream of asynchronous results, and that that is the only required API surface, but that Apollo Server includes a `PubSub` class as a convenience to implement an event bus.